[Parsers] All the Privileges to be processed prior Access artifacts

```
2021-02-24 14:00:33.867 [ERROR] [pool-8-thread-1] o.e.d.c.s.api.AbstractSynchronizer - Resource [...] skipped due to an error: com.sap.xsk.xsaccess.ds.api.XSKAccessException: com.sap.xsk.xsaccess.ds.api.XSKPrivilegeException: No XSK Privilege found with name: {sap.sports.imp.api.downloadQueue::fetch}
org.eclipse.dirigible.core.scheduler.api.SynchronizationException: com.sap.xsk.xsaccess.ds.api.XSKAccessException: com.sap.xsk.xsaccess.ds.api.XSKPrivilegeException: No XSK Privilege found with name: {a.b.c.downloadQueue::fetch}
	at com.sap.xsk.xsaccess.ds.synchronizer.XSKSecuritySynchronizer.synchronizeAccess(XSKSecuritySynchronizer.java:256) ~[xsk-modules-engines-core-0.2.0-SNAPSHOT.jar:na]
	at com.sap.xsk.xsaccess.ds.synchronizer.XSKSecuritySynchronizer.synchronizeResource(XSKSecuritySynchronizer.java:295) ~[xsk-modules-engines-core-0.2.0-SNAPSHOT.jar:na]
	at org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer.synchronizeCollection(AbstractSynchronizer.java:109) ~[dirigible-core-scheduler-5.8.1.jar:na]
```

--- a/modules/engines/engine-xssecurity/src/main/java/com/sap/xsk/xsaccess/ds/synchronizer/XSKSecuritySynchronizer.java
+++ b/modules/engines/engine-xssecurity/src/main/java/com/sap/xsk/xsaccess/ds/synchronizer/XSKSecuritySynchronizer.java
@@ -50,6 +50,12 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
   private static final Map<String, XSKAccessDefinition> ACCESS_PREDELIVERED = Collections
       .synchronizedMap(new HashMap<>());
 
+  private static final Set<XSKPrivilegeDefinition> PRIVILEGES_TO_BE_PROCESSED = Collections.synchronizedSet(
+      new HashSet<>());
+
+  private static final Set<XSKAccessDefinition> ACCESS_TO_BE_PROCESSED = Collections
+      .synchronizedSet(new HashSet<>());
+
   private static final Set<String> PRIVILEGES_SYNCHRONIZED = Collections.synchronizedSet(new HashSet<>());
 
   private static final Set<String> ACCESS_SYNCHRONIZED = Collections.synchronizedSet(new HashSet<>());
@@ -91,6 +97,7 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
 
           int mutablePrivilegesCount = PRIVILEGES_SYNCHRONIZED.size();
           int mutableAccessCount = ACCESS_SYNCHRONIZED.size();
+          processArtifacts();
           cleanup();
           clearCache();
           successfulSynchronization(SYNCHRONIZER_NAME,
@@ -157,7 +164,9 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
    */
   private void clearCache() {
     PRIVILEGES_SYNCHRONIZED.clear();
+    PRIVILEGES_TO_BE_PROCESSED.clear();
     ACCESS_SYNCHRONIZED.clear();
+    ACCESS_TO_BE_PROCESSED.clear();
     xskAccessCoreService.clearCache();
   }
 
@@ -186,6 +195,21 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
   }
 
   /**
+   * Process the privilege and access artifacts.
+   *
+   * @throws SynchronizationException the synchronization exception
+   */
+  private void processArtifacts() throws SynchronizationException {
+    for (XSKPrivilegeDefinition xskPrivilegeDefinition : PRIVILEGES_TO_BE_PROCESSED) {
+      synchronizePrivilege(xskPrivilegeDefinition);
+    }
+
+    for (XSKAccessDefinition xskAccessDefinition : ACCESS_TO_BE_PROCESSED) {
+      synchronizeAccess(xskAccessDefinition);
+    }
+  }
+
+  /**
    * Synchronize privileges.
    *
    * @param xskPrivilegeDefinition the privilege definition
@@ -206,7 +230,7 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
 
       }
 
-      ACCESS_SYNCHRONIZED.add(xskPrivilegeDefinition.getName());
+      PRIVILEGES_SYNCHRONIZED.add(xskPrivilegeDefinition.getName());
     } catch (XSKPrivilegeException e) {
       throw new SynchronizationException(e);
     }
@@ -269,7 +293,7 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
     if (resourceName.equals(IXSKPrivilegeCoreService.XSK_FILE_EXTENSION_PRIVILEGE)) {
       XSKPrivilegeArtifact xskPrivilegeArtifact = XSKPrivilegeArtifact.parse(resource.getContent());
       for (XSKPrivilegeDefinition xskPrivilegeDefinition : xskPrivilegeArtifact.divide()) {
-        synchronizePrivilege(xskPrivilegeDefinition);
+        PRIVILEGES_TO_BE_PROCESSED.add(xskPrivilegeDefinition);
       }
 
     }
@@ -280,7 +304,7 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
       String hash = DigestUtils.md5Hex(resource.getContent());
       xskAccessDefinition.setHash(hash);
 
-      synchronizeAccess(xskAccessDefinition);
+      ACCESS_TO_BE_PROCESSED.add(xskAccessDefinition);
     }
   }
 
@@ -295,7 +319,7 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
     try {
       List<XSKPrivilegeDefinition> privilegeDefinitions = xskPrivilegeCoreService.getXSKPrivileges();
       for (XSKPrivilegeDefinition xskPrivilegeDefinition : privilegeDefinitions) {
-        if (!ACCESS_SYNCHRONIZED.contains(xskPrivilegeDefinition.getName())) {
+        if (!PRIVILEGES_SYNCHRONIZED.contains(xskPrivilegeDefinition.getName())) {
 
           xskPrivilegeCoreService.removeXSKPrivilegeByName(xskPrivilegeDefinition.getName());
           logger.warn("Cleaned up XSK Privilege [{}]", xskPrivilegeDefinition.getName());
@@ -306,7 +330,8 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
       for (XSKAccessDefinition xskAccessDefinition : accessDefinitions) {
         if (!ACCESS_SYNCHRONIZED.contains(xskAccessDefinition.getPath())) {
           xskAccessCoreService.removeXSKAccessDefinition(xskAccessDefinition.getPath());
-          logger.warn("Cleaned up XSK Access definition [[{}]-[{}]-[{}]] from location: {}", xskAccessDefinition.getAuthenticationMethodsAsList(),
+          logger.warn("Cleaned up XSK Access definition [[{}]-[{}]-[{}]] from location: {}",
+              xskAccessDefinition.getAuthenticationMethodsAsList(),
               String.join(",", xskAccessDefinition.getAuthorizationRolesAsList()), xskAccessDefinition.getPath());
         }
       }

--- a/modules/engines/engine-xssecurity/src/main/java/com/sap/xsk/xsaccess/ds/synchronizer/XSKSecuritySynchronizer.java
+++ b/modules/engines/engine-xssecurity/src/main/java/com/sap/xsk/xsaccess/ds/synchronizer/XSKSecuritySynchronizer.java
@@ -25,12 +25,14 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 import org.apache.commons.codec.digest.DigestUtils;
 import org.apache.commons.io.IOUtils;
 import org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer;
@@ -292,10 +294,15 @@ public class XSKSecuritySynchronizer extends AbstractSynchronizer {
     String resourceName = resource.getName();
     if (resourceName.equals(IXSKPrivilegeCoreService.XSK_FILE_EXTENSION_PRIVILEGE)) {
       XSKPrivilegeArtifact xskPrivilegeArtifact = XSKPrivilegeArtifact.parse(resource.getContent());
-      for (XSKPrivilegeDefinition xskPrivilegeDefinition : xskPrivilegeArtifact.divide()) {
-        PRIVILEGES_TO_BE_PROCESSED.add(xskPrivilegeDefinition);
-      }
 
+      // set name to use "path.to::Privilege" syntax
+      String[] splitPath = resource.getPath().split("/");
+      String path = String.join(".", Arrays.copyOfRange(splitPath, 3, splitPath.length - 1)); // remove /registry/public
+
+      for (XSKPrivilegeDefinition privilege : xskPrivilegeArtifact.divide()) {
+        privilege.setName(path + "::" + privilege.getName());
+        PRIVILEGES_TO_BE_PROCESSED.add(privilege);
+      }
     }
     if (resourceName.equals(IXSKAccessCoreService.XSK_FILE_EXTENSION_ACCESS)) {
       XSKAccessDefinition xskAccessDefinition = xskAccessCoreService.parseXSAccessArtifact(resource.getContent());

